src/commands/compile/compileSingleFile.ts

```diff
@@ -0,0 +1,86 @@
+import path from 'path'
+import { getProgramFolders } from '../../utils/config'
+import { copy, fileExists, deleteFolder, createFolder } from '../../utils/file'
+import { Target } from '@sasjs/utils/types'
+import { compileServiceFile } from './internal/compileServiceFile'
+import { compileJobFile } from './internal/compileJobFile'
+import { Command } from '../../utils/command'
+
+export async function compileSingleFile(
+  target: Target,
+  command: Command,
+  subCommand: string
+) {
+  const subCommands = {
+    job: 'job',
+    service: 'service'
+  }
+
+  if (!subCommands.hasOwnProperty(subCommand)) {
+    throw new Error(
+      `Unsupported context command. Supported commands are:\n${Object.keys(
+        subCommands
+      ).join('\n')}`
+    )
+  }
+
+  const commandExample =
+    'sasjs compile <command> --source myjob.sas --target targetName -output /some/folder'
+
+  const source = command.getFlagValue('source') as string
+  const output = command.getFlagValue('output') as string
+
+  if (!source) {
+    throw new Error(`'--source' flag is missing (eg '${commandExample}')`)
+  } else if (!(await validateSourcePath(source))) {
+    throw new Error(`Provide a path to source file (eg '${commandExample}')`)
+  }
+
+  const sourcePath = path.resolve(process.projectDir, source)
+  const outputPath = path.resolve(process.projectDir, output ?? './sasjsbuild')
+
+  process.logger?.info(`Compiling source file:\n- ${sourcePath}`)
+
+  await deleteFolder(outputPath)
```

Comment:
in `sasjsbuild` can be some data, why do we need to delete it?

Reply:
if we don't delete the previous content, then it's possible that `sasjs build` will attempt to create build files (.json / .sas) for a previous (project wide) `sasjs compile` - meaning that we couldn't subsequently / consistently `sasjs deploy` a single file

Reply:
but `sasjsbuild` is also a default location for such files as logs (if another location wasn't provided).

Reply:
correct - all of these items are temporary / transient (not for source control).  Similar to the 'dist' folder (wiped when doing a frontend build).

I do wonder if it would be better if we renamed it as the `sasjstemp` folder.

Reply:
if the user wants to keep the logs, they can move them afterwards, or redirect them somewhere permanent to begin with